facefusion/processors/frame/modules/lip_sync.py

```diff
@@ -0,0 +1,222 @@
+
+	if OPTIONS is None:
+		OPTIONS =\
+		{
+			'model': MODELS[frame_processors_globals.lip_sync_model]
+		}
+	return OPTIONS.get(key)
+
+
+def set_options(key : Literal['model'], value : Any) -> None:
+	global OPTIONS
+
+	OPTIONS[key] = value
+
+
+def register_args(program : ArgumentParser) -> None:
+	program.add_argument('--lip-sync-model', help = wording.get('help.lip_sync_model_help'), default = config.get_str_value('frame_processors.lip_sync_model', 'wav2lip_gan'), choices = frame_processors_choices.lip_sync_models)
+
+
+def apply_args(program : ArgumentParser) -> None:
+	args = program.parse_args()
+	frame_processors_globals.lip_sync_model = args.lip_sync_model
+
+
+def pre_check() -> bool:
+	if not facefusion.globals.skip_download:
+		download_directory_path = resolve_relative_path('../.assets/models')
+		model_url = get_options('model').get('url')
+		conditional_download(download_directory_path, [ model_url ])
+	return True
+
+
+def post_check() -> bool:
+	model_url = get_options('model').get('url')
+	model_path = get_options('model').get('path')
+	if not facefusion.globals.skip_download and not is_download_done(model_url, model_path):
+		logger.error(wording.get('model_download_not_done') + wording.get('exclamation_mark'), NAME)
+		return False
+	elif not is_file(model_path):
+		logger.error(wording.get('model_file_not_present') + wording.get('exclamation_mark'), NAME)
+		return False
+	return True
+
+
+def pre_process(mode : ProcessMode) -> bool:
+	audio_paths = filter_audio_paths(facefusion.globals.source_paths)
+	if not audio_paths:
+		logger.error(wording.get('select_audio_source') + wording.get('exclamation_mark'), NAME)
+		return False
+	if mode in [ 'output', 'preview' ] and not is_video(facefusion.globals.target_path):
+		logger.error(wording.get('select_video_target') + wording.get('exclamation_mark'), NAME)
+		return False
+	if mode == 'output' and not facefusion.globals.output_path:
+		logger.error(wording.get('select_file_or_directory_output') + wording.get('exclamation_mark'), NAME)
+		return False
+	return True
+
+
+def post_process() -> None:
+	clear_frame_processor()
+	clear_face_analyser()
+	clear_content_analyser()
+	clear_face_occluder()
+	clear_face_parser()
+	read_static_audio.cache_clear()
+
+
+def lip_sync(audio_frame : AudioFrame, target_face : Face, temp_frame : VisionFrame) -> VisionFrame:
+	frame_processor = get_frame_processor()
+	crop_frame, affine_matrix = warp_face_by_bbox(temp_frame, target_face.bbox, (96, 96))
+	audio_frame = prepare_audio_frame(audio_frame)
+	crop_frame = prepare_crop_frame(crop_frame)
+	crop_frame = frame_processor.run(None, {'vid' : crop_frame, 'mel' : audio_frame})[0]
+	crop_frame = normalize_crop_frame(crop_frame)
+	crop_mask = create_static_box_mask(crop_frame.shape[:2][::-1], 0.1, (50, 0, 0, 0))
+	paste_frame = paste_back(temp_frame, crop_frame, crop_mask, affine_matrix)
+	if 'occlusion' in facefusion.globals.face_mask_types or 'region' in facefusion.globals.face_mask_types:
```

Comment:
I do not like the nested conditions...

1. move `crop_mask_list = []` out of the condition
2. fill it up according to occlussion or region is active
3. only process/paste back if the crop_mask_list has value